src/lando/main/models/jobs.py

```diff
@@ -275,14 +275,20 @@ def to_dict(self) -> dict[str, Any]:
         """Return the job details as a dict."""
         job_dict = {
             "created_at": self.created_at,
-            "error": self.error,
+            "error": None,
             "id": self.id,
             "requester": self.requester_email,
             "status": self.status,
             "updated_at": self.updated_at,
         }
 
+        if self.status in [JobStatus.FAILED, JobStatus.DEFERRED]:
+            job_dict["error"] = self.error
```

Comment:
Why not add the `error` field anyway, to make the response body more consistent?